### DISCUSSION TITLE
Version 2.1.0 ImageHelper.php Error

### DISCUSSION THREAD

**Discussion by shandyw** [flyntwp/flynt#547] [Q&A]
After cloning the repo and doing the install commands, I immediately get an error. 

Warning: Undefined array key 0 in /Users/me/Sites/valet/bp/web/app/themes/flynt/vendor/timber/timber/src/ImageHelper.php on line 444

Deprecated: rawurlencode(): Passing null to parameter #1 ($string) of type string is deprecated in /Users/me/Sites/valet/bp/web/app/themes/flynt/vendor/timber/timber/src/PathHelper.php on line 75

Fatal error: Uncaught ValueError: Path cannot be empty in /Users/me/Sites/valet/bp/web/app/themes/flynt/vendor/timber/timber/src/ImageHelper.php:452 

I know it's caused by something in inc/sideLoadAvatarImages.php. 
(Theme works if it is deleted).

I'm using PHP 8.1 btw.

I'm new to Timber, can anyone help? I'd love to fix it. TY!

**Comment by timohubois** [comment]
@shandyw thanks for starting the discussion.

The problem seems to be related to image urls that do not contain a file extension and it is related to Timber. They have already [fixed it](https://github.com/timber/timber/commit/028f6ac0bb2458823d9baaae34da46657e8fb6d1) and I hope they will release the new version soon.

We added this functionality in the context of GDPR to ensure that avatar images are loaded locally. Deleting the file should not break the website and will bring back the native behavior how avatar image urls are handled by WordPress.

What is currently not reproducible for me is the fatal error. I also can reproduce the warning and testing with my setup (PHP 8.0, 8.1, 8.2 and 8.3) does break the site nor throws an error – instead I get the warning **once** (when the avatar was not downloaded before) and from the distance it looks like that the error will thrown when the avatar image can or is not stored locally.
